File: Problem_16/solution_16.lua
-- Euler Project 16
-- Task: 2^15 = 32768 and the sum of its digits is 3 + 2 + 7 + 6 + 8 = 26.
-- What is the sum of the digits of the number 2^1000?

require "math"

-- This number is too lage for the lua standard number range --> Python it is
a=2^1000

-- Calculating the sum of the digits
summe = 0
for i=1, string.len(a) do
	summe = summe + tonumber(string.sub(a, i, i))
end

print("The sum of the digits is " .. summe)

-- Solution: 1366 (correct)

File: Problem_16/solution_16.py
# Euler Project 16
# Source: https://projecteuler.net/problem=16
import time
start_time = time.time()

a,summe = str(2**1000), 0
for i in range(0, len(a)):
	summe += int(a[i])
	
print(summe)
print("The script took %s seconds for execution" % (time.time() - start_time))

# Solution: 1366 (correct)
# Runtime: <0.0 seconds